homeassistant/components/forked_daapd/media_player.py

```diff
@@ -0,0 +1,669 @@
+"""This library brings support for forked_daapd to Home Assistant."""
+import asyncio
+from collections import defaultdict
+import logging
+
+from pyforked_daapd import ForkedDaapdAPI
+from pylibrespot_java import LibrespotJavaAPI
+
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.components.media_player.const import MEDIA_TYPE_MUSIC
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_ON,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    CONF_DEFAULT_VOLUME,
+    CONF_PIPE_CONTROL,
+    CONF_PIPE_CONTROL_PORT,
+    CONF_TTS_PAUSE_TIME,
+    CONF_TTS_VOLUME,
+    DEFAULT_TTS_PAUSE_TIME,
+    DEFAULT_TTS_VOLUME,
+    DOMAIN,
+    FD_NAME,
+    HASS_DATA_MASTER_KEY,
+    HASS_DATA_OUTPUTS_KEY,
+    HASS_DATA_REMOVE_ENTRY_LISTENER_KEY,
+    SERVER_UNIQUE_ID,
+    SIGNAL_ADD_ZONES,
+    SIGNAL_UPDATE_MASTER,
+    SIGNAL_UPDATE_OUTPUTS,
+    SIGNAL_UPDATE_PLAYER,
+    SIGNAL_UPDATE_QUEUE,
+    STARTUP_DATA,
+    SUPPORTED_FEATURES,
+    SUPPORTED_FEATURES_ZONE,
+    TTS_TIMEOUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+WS_NOTIFY_EVENT_TYPES = ["player", "outputs", "volume", "options", "queue"]
+WEBSOCKET_RECONNECT_TIME = 30  # seconds
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up forked-daapd from a config entry."""
+    name = config_entry.data.get(CONF_NAME)
+    host = config_entry.data.get(CONF_HOST)
```

Comment:
OK, will try to keep an eye on it. Sometimes hard to keep track of what is there or not.
BTW, since everything in the config has defaults should the vol schema entries be changed from optional to required?

Reply:
No, optional means that the user doesn't have to fill those items in.